pallets/pools/src/weights.rs

```diff
@@ -56,101 +49,90 @@ pub trait WeightInfo {
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
-	fn create(_n: u32) -> Weight {
-		(107_116_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+	fn create(n: u32) -> Weight {
+		(74_485_000 as Weight) // Standard Error: 41_000
+			.saturating_add((299_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(6 as Weight))
+			.saturating_add(T::DbWeight::get().writes(6 as Weight))
 	}
-	fn update() -> Weight {
-		(59_656_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(2 as Weight))
+	fn update_no_execution(n: u32) -> Weight {
+		(28_660_000 as Weight) // Standard Error: 17_000
+			.saturating_add((285_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(4 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
+	fn update_and_execute(n: u32) -> Weight {
+		(47_893_000 as Weight) // Standard Error: 40_000
+			.saturating_add((716_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(4 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+	}
+	fn execute_scheduled_update(n: u32) -> Weight {
+		(45_439_000 as Weight) // Standard Error: 64_000
+			.saturating_add((1_074_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(3 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+	}
 	fn set_metadata(n: u32) -> Weight {
-		(58_225_000 as Weight)
-			// Standard Error: 37_000
-			.saturating_add((14_000 as Weight).saturating_mul(n as Weight))
+		(35_072_000 as Weight) // Standard Error: 0
+			.saturating_add((13_000 as Weight).saturating_mul(n as Weight))
 			.saturating_add(T::DbWeight::get().reads(2 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn set_max_reserve() -> Weight {
-		(58_896_000 as Weight)
+		(34_009_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(2 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
-	fn update_no_execution() -> Weight {
-		(61_909_000 as Weight)
-			// Standard Error: 632_000
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
-	}
-	fn update_and_execute() -> Weight {
-		(61_909_000 as Weight)
-			// Standard Error: 632_000
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
-	}
-	fn execute_scheduled_update() -> Weight {
-		(61_909_000 as Weight)
-			// Standard Error: 632_000
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
-	}
 	fn update_invest_order() -> Weight {
-		(169_699_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(8 as Weight))
+		(100_107_000 as Weight)
```

Comment:
Those numbers were from my laptop, I only ran the per-runtime weights on the bench host.

Reply:
And to be clear these per-pallet numbers only get used in the tests. All the runtimes have their own benches.

It's probably worth re-running the development runtime benches in the bench env instead of my laptop (since apparently my laptop is more powerful), but I don't think it's worth blocking this.